### PR TITLE
Bento: document-scoped utils

### PR DESCRIPTION
**Pull request by scottrippey**
What:
- Adds a `OwnerDocumentContext` to all Bento components.  
  - In Preact mode, this defaults to the global `document`
  - In Bento / Amp modes, a Provider will automatically be added with the value set to `element.ownerDocument`
  - A hook can be used to access the value; eg `const ownerDocument = useOwnerDocument();`
- Eliminates access to global `document` within Bento utilities (eg. `urlUtils` and `docInfoUtils` both used `self.document`).
  - Instead, these utils have been refactored to require an `ownerDocument` 
  - The same pattern has been used to ensure `amp-app-banner` methods are tied to an `ownerDocument`

### Example usage:
Before:
```
function BentoComponent({ }) {
  const parsedUrl = urlUtils.parse('https://example.com');
  // ... etc ...
}
```

After:
```
function BentoComponent({ }) {
  const ownerDocument = useOwnerDocument();
  const parsedUrl = UrlUtils.forDoc(ownerDocument).parse('https://example.com');
  // ... etc ...
}
```

### Notes:
Several syntax variations were considered for passing around the `ownerDocument`.  Ultimately, the `forDoc(doc)` approach was the most ergonomic and semantic, and works well for the majority of use-cases.

### CODE REVIEW

File: extensions/amp-app-banner/1.0/component/component.js [hunk 1]
```diff
@@ -73,7 +74,10 @@ export function AppBanner({
  * @constructor
  */
 function AppBannerIOS(props) {
-  const appInfo = useMemo(getIOSAppInfo, []);
+  const ownerDocument = useOwnerDocument();
```
Comment (becca-bailey) [inline extensions/amp-app-banner/1.0/component/component.js]: Nice! I will need to do the same for my component.

**Comment by dethstrobe** [comment]
This looks like a PR in to your own repo?

**Comment by scottrippey** [comment]
Yes, just to get early feedback on the approach.  This PR is dependent on `bento-list` getting merged first, so I didn't bother creating an "official" PR yet.

**Comment by scottrippey** [comment]
Creating an official PR now.

**Pull request closed without merging**